### PR TITLE
Refactor Get/Create Direct Channel into one function

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
- Refactor Get/Create Direct Channel into one function
- removed the CreateDirectChannel plugin api and use the GetDirectChannel plugin api (also update the comments)

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by cpanato** [comment]
@lieut-data renamed to GetOrCreate which makes more sense

File: app/channel.go [hunk 1]
```diff
@@ -224,35 +224,42 @@ func (a *App) CreateChannel(channel *model.Channel, addMember bool) (*model.Chan
...
+			channel, err := a.createDirectChannel(userId1, userId2)
+			if err != nil {
+				if err.Id == store.CHANNEL_EXISTS_ERROR {
+					return channel, nil
+				}
+				return nil, err
+			}
 
-	a.WaitForChannelMembership(channel.Id, userId)
+			a.WaitForChannelMembership(channel.Id, userId1)
```
[the remark this answers is not in the document]
Comment (jwilander) [reply app/channel.go]: I can't remember the specifics, but IIRC it's dealing with a race condition related to HA where on some servers the in-memory cache doesn't have the new channel member yet and this was causing some issues. Of course, it was meant as a quick patch to the issue at the time and has yet to be revisited

[the remark this answers is not in the document]
Comment (jwilander) [reply app/channel.go]: +1 on going with `userId` and `otherUserId` though

[the remark this answers is not in the document]
Comment (cpanato) [reply app/channel.go]: done the rename

**Comment by jwilander** [comment]
@cpanato we should cherry-pick this one as we're removing a plugin API (before it got released)

**Comment by cpanato** [comment]
@jwilander ok, will do that

**Comment by cpanato** [comment]
cherry picked 1bcf08aa4b924ecf9425c68e55ffdc0f8edd6606 into release 5.6 branch

**Pull request merged**